mlflow/spark.py

```diff
@@ -181,6 +215,40 @@ def delete(cls, path):
         cls._fs().delete(cls._remote_path(path), True)
 
 
+def _save_model_metadata(dst_dir, spark_model, mlflow_model, sample_input, conda_env):
+    """
+    Saves model metadata into the passed-in directory. The persisted metadata assumes that a
+    model can be loaded from a relative path to the metadata file (currently hard-coded to
+    "sparkml").
+    """
+    if sample_input is not None:
+        mleap.add_to_model(mlflow_model, dst_dir, spark_model, sample_input)
+
+    pyspark_version = pyspark.version.__version__
+
+    conda_env_subpath = "conda.yaml"
+    if conda_env is None:
+        conda_env = DEFAULT_CONDA_ENV
+    elif not isinstance(conda_env, dict):
+        with open(conda_env, "r") as f:
+            conda_env = yaml.safe_load(f)
+    with open(os.path.join(dst_dir, conda_env_subpath), "w") as f:
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
+
+    mlflow_model.add_flavor(FLAVOR_NAME, pyspark_version=pyspark_version,
+                            model_data=_SPARK_MODEL_PATH_SUB)
+    pyfunc.add_to_model(mlflow_model, loader_module="mlflow.spark", data=_SPARK_MODEL_PATH_SUB,
+                        env=conda_env_subpath)
+    mlflow_model.save(os.path.join(dst_dir, "MLmodel"))
+
+
+def _validate_model(spark_model, jars):
+    if not isinstance(spark_model, PipelineModel):
+        raise Exception("Not a PipelineModel. SparkML can only save PipelineModels.")
```

Comment:
Nit: Can we change these to MLflowExceptions with the `INVALID_PARAMETER_VALUE` error code?